src/Components/Components/src/Lifetime/IComponentApplicationStateStore.cs

```diff
@@ -0,0 +1,24 @@
+using System.Collections.Generic;
+using System.Threading.Tasks;
+
+namespace Microsoft.AspNetCore.Components
+{
+    /// <summary>
+    /// Manages the storage for components and services that are part of a Blazor application.
+    /// </summary>
+    public interface IComponentApplicationStateStore
```

Comment:
This is `public`, but it would never work for a customer to implement their own, since our hosting logic relies on it being our own concrete classes.

I guess that's just the same as something like `IJSRuntime` so maybe it's fine. But if there's some way of making it not `public`, that would be easier to understand.

Reply:
That is correct but in the (near) future, when we look at pausing and resuming circuits, I expect this to be the thing customers implement to define their own circuit storage mechanism.

That said, I don't think this can be made private due to the layering (MVC needs to be able to consume it)

Reply:
Do you think we could consider making it public when we're ready to do so? For now, does IVT work?

Reply:
Do you have a specific concern about it? This is not something that a user will be able to use and it's not in their faces, akin to Renderer.

We don't have IVT anywhere and I would prefer if we avoid it, that said I take note to revisit this in the future as we do more work in this area.